### PR TITLE
[IMP] l10n_fr_reports: Round lines of tax report and closing entries to the nearest euro

### PR DESCRIPTION
**Pull request by dbkosky**
French tax reports require that each line is rounded to the nearest euro.
 In the l10n_fr localisation add default rounding accounts to the chart template and the company, add them to the try loading section of account chart template data xml

== PRs ==
odoo/enterprise#20245

task: 2610752

### CODE REVIEW

File: addons/l10n_fr/data/account_chart_template_data.xml
Comment (qdp-odoo) [inline addons/l10n_fr/data/account_chart_template_data.xml:25]: field name should be prefixed with l10n_fr, because it's the guideline for fields used only in localizations

File: addons/l10n_fr/models/res_company.py
Comment (qdp-odoo) [inline addons/l10n_fr/models/res_company.py:14]: idem

**Review by qdp-odoo — CHANGES_REQUESTED** [review]

**Review by qdp-odoo — APPROVED** [review]
only a small comment

File: addons/l10n_fr/models/account_chart_template.py
Comment (william-andre) [inline addons/l10n_fr/models/account_chart_template.py:26]: Why only if the company's country code is `== 'FR'`?
If a company in another country wants to load the chart template, so be it :shrug: 
For instance, this is a country code I would expect uses the same chart template and roundings https://www.iso.org/obp/ui/#iso:code:3166:MQ

File: addons/l10n_fr/models/account_chart_template.py
Comment (qdp-odoo) [reply addons/l10n_fr/models/account_chart_template.py:26]: @dbkosky this is still missing an `IF company.country_id.code == 'FR'`

**Comment by dbkosky** [comment]
@qdp-odoo just stating it here, I've re-removed the conditional on loading the fields.
The reason is that I agree with William's comment (https://github.com/odoo/odoo/pull/75634#discussion_r859212091). Writing to the fields is only attempted if we're loading the French CoA, and if we're loading the French CoA, why not use these accounts in these fields?

File: addons/l10n_fr/models/account_chart_template.py
Comment (william-andre) [reply addons/l10n_fr/models/account_chart_template.py:26]: You need to check on the chart template itself, not on the company's country.
For instance, I expect Martinique to use the same CoA, but the country code is MQ.

**Comment by antoine162** [comment]
Hiiii @dbkosky I was just about to look at this (sorry for the long wait, I didn't realise this was waiting to be reviewed)
but I assume something went wrong during your last rebase? :wink: ?

**Comment by tsb-odoo** [comment]
@antoine162 Is this PR ready to be merged ? Anything needed ?

**Comment by dbkosky** [comment]
@tsb-odoo It's waiting on @jco-odoo 's approval

**Comment by jco-odoo** [comment]
@robodoo r+

**Pull request closed without merging**
